### DISCUSSION TITLE
Debug with breakpoint

### DISCUSSION THREAD

**Discussion by manish2612** [software-mansion/react-native-reanimated#1816] [Q&A] [answered]
As per limitation of Reanimated-2 debugging is not an option unless we use Flipper but Flipper doesn't support breakpoint in JS code. Is there any way we can use Reanimated-2 and use breakpoints in JS code?

**Comment by mrousavy** [comment]
Why does Flipper not support JS breakpoints? I haven't tried it yet, but I was under the impression that that was already working fine 🤔  (with latest flipper 0.81 & Hermes)

**Reply by mrousavy** [reply]
Note that debugging the worklets is not possible. I'm only talking about the normal React-JS runtime code.

**Comment by terrysahaidak** [comment, marked as answer]
I think you should be able to debug js code with breakpoints if you attach to hermes directly.

https://reactnative.dev/docs/hermes#debugging-js-on-hermes-using-google-chromes-devtools

You can track progress on Hermes debugging here: https://github.com/microsoft/vscode-react-native/issues/1266

You can also attach to the JSC process using Safari too. Also, you can attach to the Reanimated worklets context using Safari too.
